src/idl_gen_rust.cpp

```diff
@@ -737,7 +737,7 @@ class RustGenerator : public BaseGenerator {
       code_ += "pub use self::bitflags_{{ENUM_NAMESPACE}}::{{ENUM_TY}};";
       code_ += "";
 
-      code_.SetValue("FROM_BASE", "unsafe { Self::from_bits_unchecked(b) }");
+      code_.SetValue("FROM_BASE", "Self::from_bits_truncate(b)");
```

Comment:
I couldn't devise a plausible SAFETY comment for this, so switched to using the safe version of the API. The performance impact will be imperceptible.

Reply:
The issue with `from_bits_truncate` is that it breaks the Flatbuffers contract - we don't truncate bits in other languages.

The bitflags developer has reconsidered marking this function unsafe. It doesn't lead to memory unsafety. See the follwoing context:  https://github.com/bitflags/bitflags/issues/262

Reply:
What do you suggest in that case?

Reply:
```rs
// Safety:
// This is safe because we know bitflags is implemented with a repr transparent uint of the correct size.
// from_bits_unchecked will be replaced by an equivalent but safe from_bits_retain in bitflags 2.0
// https://github.com/bitflags/bitflags/issues/262
```